instances/expander.go

```diff
@@ -150,16 +150,14 @@ func (e *Expander) ExpandModuleResource(moduleAddr addrs.Module, resourceAddr ad
 // itself must already have had their expansion registered using one of the
 // SetModule*/SetResource* methods before calling, or this method will panic.
 func (e *Expander) ExpandResource(resourceAddr addrs.AbsResource) []addrs.AbsResourceInstance {
-	var ret []addrs.AbsResourceInstance
+	e.mu.RLock()
+	defer e.mu.RUnlock()
 
-	// Take the full set of expanded resource instances and filter for only
-	// those within our module instance.
-	for _, r := range e.ExpandModuleResource(resourceAddr.Module.Module(), resourceAddr.Resource) {
-		if !r.Module.Equal(resourceAddr.Module) {
-			continue
-		}
-		ret = append(ret, r)
-	}
+	moduleInstanceAddr := make(addrs.ModuleInstance, 0, 4)
+	ret := e.exps.resourceInstances(resourceAddr.Module, resourceAddr.Resource, moduleInstanceAddr)
+	sort.SliceStable(ret, func(i, j int) bool {
```

Comment:
Why are these getting sorted?

Reply:
Just for consistency, so that things like output and test fixtures are stable.
It also matched the behavior of the original method above, now called `ExpandModuleResource`.